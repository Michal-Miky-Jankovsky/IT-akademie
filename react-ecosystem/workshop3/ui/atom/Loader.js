import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import '~/ui/atom/loader.css';

export default function Loader({mode, layout}) {
  return (
    <div
      className={classnames(
        'atm-loader',
        {
          [`atm-loader-${mode}`]: mode,
          [`atm-loader-${layout}`]: layout,
        },
      )}
    >
      <span/>
      <span/>
      <span/>
      <span/>
      <span/>
    </div>
  );
}

Loader.propTypes = {
  mode: PropTypes.oneOf(['small']),
  layout: PropTypes.oneOf(['center']),
};
