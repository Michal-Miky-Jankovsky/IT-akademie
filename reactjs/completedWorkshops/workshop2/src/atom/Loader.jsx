import React, { PropTypes } from 'react';
import { uiComponentHelper } from '../UIComponentHelper';

/**
 * Common loader
 *
 * @class Loader
 */

export default function Loader(props) {
	return (
		<div className = { uiComponentHelper.cssClasses({
			'atm-loader': true,
			['atm-loader-' + props.mode]: props.mode,
			['atm-loader-' + props.layout]: props.layout
		}, props.className) }>
			<span/>
			<span/>
			<span/>
			<span/>
			<span/>
		</div>
	);
}

Loader.propTypes = {
	mode: PropTypes.string,
	layout: PropTypes.string,
	className: PropTypes.string
};

Loader.defaultProps = {
	mode: '',
	layout: '',
	className: ''
};
