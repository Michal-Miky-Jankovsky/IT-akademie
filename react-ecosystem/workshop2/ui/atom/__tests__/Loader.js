import React from 'react';
import Renderer from 'react-test-renderer';
import Loader from '~/ui/atom/Loader';

describe('Loader', () => {
  it('should render correctly with no props', () => {
    const domTree = Renderer.create(<Loader/>).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it('should render correctly with mode and layout passed in props', () => {
    const domTree = Renderer.create(<Loader mode="small" layout="center"/>).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
