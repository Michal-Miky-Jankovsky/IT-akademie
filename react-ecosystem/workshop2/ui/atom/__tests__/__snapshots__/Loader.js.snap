// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly with mode and layout passed in props 1`] = `
<div
  className="atm-loader atm-loader-small atm-loader-center"
>
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`Loader should render correctly with no props 1`] = `
<div
  className="atm-loader"
>
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;
